tests/test_visualise_results.py

```diff
@@ -3,60 +3,160 @@
-    assert_frame_equal(dataframe, expected_data_frame_multiple_files)
 
+@pytest.mark.parametrize(
+    "folders, data_frame_method, expected_result",
+    (
+        (
+            [
+                "./tests/mock_stats/2024-02-07T03:09:41",
+                "./tests/mock_stats/2024-02-06T03:09:41",
+            ],
+            get_performance_data_frame,
+            expected_data_frame_multiple_files,
+        ),
+        (
+            [
+                "./tests/mock_stats/2024-07-25T03:09:41",
+                "./tests/mock_stats/2024-07-29T03:09:41",
+            ],
+            get_additional_metrics_data_frame,
+            expected_data_frame_multiple_files_session_pdf,
+        ),
+    ),
+)
+def test_get_data_frame_multiple_files(folders, data_frame_method, expected_result):
+    results = get_results(folders=folders)
+    dataframe = data_frame_method(results)
+    assert_frame_equal(dataframe, expected_result)
 
-def test_plot_data_df(mocker, get_results_single_file):
-    dataframe = get_data_frame(get_results_single_file)
+
+@pytest.mark.parametrize(
+    "expected_dataframe, data_frame_method, results_file",
+    (
+        (expected_data_frame, get_performance_data_frame, "get_results_single_file"),
+        (
+            expected_data_frame_session_pdf,
+            get_additional_metrics_data_frame,
+            "get_results_single_file_with_pdf_endpoint",
+        ),
+    ),
+)
+def test_plot_data_df(
+    expected_dataframe, data_frame_method, results_file, mocker, request
+):
+    dataframe = data_frame_method(request.getfixturevalue(results_file))
 
     mock_plot_data = mocker.patch("scripts.visualise_results.plot_data")
     mock_plot_data(dataframe, 1)
 
     assert mock_plot_data.call_count == 1
-    assert_frame_equal(mock_plot_data.call_args[0][0], expected_data_frame)
+    assert_frame_equal(mock_plot_data.call_args[0][0], expected_dataframe)
 
 
-def test_plot_data(mocker, get_results_single_file):
-    dataframe = get_data_frame(get_results_single_file)
+@pytest.mark.parametrize(
+    "results_file, data_frame_method, image_name",
+    (
+        (
+            "get_results_single_file",
+            get_performance_data_frame,
+            "performance_graph.png",
+        ),
+        (
+            "get_results_single_file_with_pdf_endpoint",
+            get_additional_metrics_data_frame,
+            "additional_metrics.png",
+        ),
+    ),
+)
+def test_individual_graph_creation(
+    results_file, data_frame_method, image_name, mocker, request
+):
+    dataframe = data_frame_method(request.getfixturevalue(results_file))
     try:
         graph_output = mocker.patch("matplotlib.pyplot.savefig")
-        plot_data(dataframe, 1)
+        create_graph([dataframe], 1, image_name)
         assert graph_output.call_count == 1
     except GraphGenerationFailed:
         pytest.fail("Graph generation failed")
 
 
-def test_plot_data_failed():
+@pytest.mark.parametrize(
+    "image_name",
```

Comment:
Is there any point parametrising these? They are just strings

Reply:
I guess this makes sense so you don't need to create the for loop in the test?

Reply:
That was the reason why I did it, like Pete said, however, I do understand that there's no point in have the image name parametrised since the filename could be anything, so we could have loads of filenames and it won't make a difference to the test

Reply:
@petechd But the actual value doesn't matter to the test in this instance, it is just a string and could be anything?